python/dolfinx/plotting.py

```diff
@@ -439,3 +438,78 @@ def plot(object, *args, **kwargs):
         return _plot_matplotlib(object, mesh, kwargs)
     else:
         assert False, "This code should not be reached."
+
+
+# Permutation for Dolfinx DG layout to VTK
+# Note that third order tetrahedrons has a special ordering: https://gitlab.kitware.com/vtk/vtk/-/issues/17746
+_perm_dg = {cpp.mesh.CellType.triangle: {1: [0, 1, 2], 2: [0, 2, 5, 1, 4, 3], 3: [0, 3, 9, 1, 2, 6, 8, 7, 4, 5],
+                                         4: [0, 4, 14, 1, 2, 3, 8, 11, 13, 12, 9, 5, 6, 7, 10]},
+            cpp.mesh.CellType.tetrahedron: {1: [0, 1, 2, 3], 2: [0, 2, 5, 9, 1, 4, 5, 6, 7, 8],
+                                            3: [0, 3, 9, 19, 1, 2, 6, 8, 7, 4, 10, 16, 12, 17, 15, 18, 11, 14, 13, 5]}}
+_perm_dq = {cpp.mesh.CellType.quadrilateral: {1: [0, 1, 3, 2], 2: [0, 2, 8, 6, 1, 5, 7, 3, 4],
+                                              3: [0, 3, 15, 12, 1, 2, 7, 11, 13, 14, 4, 8, 5, 6, 9, 10]},
+            cpp.mesh.CellType.hexahedron: {1: [0, 1, 3, 2, 4, 5, 7, 6],
+                                           2: [0, 2, 8, 6, 18, 20, 26, 24, 1, 5, 7, 3, 19,
+                                               23, 25, 21, 9, 11, 17, 15, 12, 14, 10, 16, 4, 22, 14]}}
+
+# NOTE: Edge visualization of higher order elements are sketchy, see:
+# https://github.com/pyvista/pyvista/issues/947
+
+
+def pyvista_topology_from_mesh(mesh, dim, entities=None):
+    """
+    Create a pyvista mesh based on a set of entities of a given dimension in a mesh
+    """
+    if entities is None:
+        num_cells = mesh.topology.index_map(dim).size_local
+        entities = np.arange(num_cells, dtype=np.int32)
+    else:
+        num_cells = len(entities)
+    geometry_entities = cpp.mesh.entities_to_geometry(mesh, dim, entities, False)
+    cell_types = np.full(num_cells, cpp.io.get_vtk_cell_type(mesh, dim))
+    num_nodes = geometry_entities.shape[1]
+    e_type = cpp.mesh.cell_entity_type(mesh.topology.cell_type, dim)
+    map_vtk = cpp.io.transpose_map(cpp.io.perm_vtk(e_type, num_nodes))
+    topology = np.zeros((num_cells, num_nodes + 1), dtype=np.int32)
+    topology[:, 0] = num_nodes
+    topology[:, 1:] = geometry_entities[:, map_vtk]
+    return topology.reshape(1, -1)[0], cell_types
+
+
+def pyvista_topology_from_function_space(u, entities=None):
```

Comment:
Looks like this function only returns mesh data?

Reply:
Yes, it only returns the topology, as we can easily access the underlying geometry (with `u.function_space.tabulate_dof_coordinates`) and the nodal data as `u.vector.array`,
I could move the  return of this data into this function. However, as there are special reshaping options for vector data, and if you would like to use the real or complex part of the function, I've let these return types out of the function

Reply:
Note that the `dofmap` used to make the topology in this function is the one of the function-space.

Reply:
It would make sense to only pass in `topology` rather than `u`.

Reply:
How would you suggest to do that?
The topology created from the function space is not the mesh topology. It is a topology based on the degrees of freedom of the function space, and the corresponding dof coordinates.
This means that if we supply a DG 1- DG N field, we create a mesh where no cell is connected to its neighbour.

For a CG-N space on a CG-1 mesh, it creates the topology of a CG-N mesh.

Reply:
I think the docstring needs updating since it sounds like it's creating a new topology that isn't based on the mesh, whereas the docstring says that it returns the mesh topology (and geometry, which I don't see in the return args).